packages/alea-frontend/components/instructor-dash.tsx

```diff
@@ -0,0 +1,325 @@
+import ArticleIcon from '@mui/icons-material/Article';
+import CommentIcon from '@mui/icons-material/Comment';
+import Diversity3Icon from '@mui/icons-material/Diversity3';
+import GradingIcon from '@mui/icons-material/Grading';
+import LockOpenIcon from '@mui/icons-material/LockOpen';
+import QuizIcon from '@mui/icons-material/Quiz';
+import { Avatar, Box, Card, CardActionArea, CardContent, Typography } from '@mui/material';
+import {
+  getCourseGradingItems,
+  getCourseQuizList,
+  getHomeworkList,
+  getUserInfo,
+} from '@stex-react/api';
+import { Action, CourseResourceAction, PRIMARY_COL, ResourceName } from '@stex-react/utils';
+import axios from 'axios';
+import dayjs from 'dayjs';
+import { useRouter } from 'next/router';
+import { useEffect, useMemo, useState } from 'react';
+import MainLayout from '../layouts/MainLayout';
+
+const excludedResources = [
+  ResourceName.COURSE_STUDY_BUDDY,
+  ResourceName.COURSE_ACCESS,
+  ResourceName.COURSE_COMMENTS,
+];
+
+function calculateTimeAgo(timestamp: string): string | null {
+  if (!timestamp) return null;
+  const dateTime = dayjs(timestamp);
+  return dateTime.isValid() ? dateTime.fromNow() : null;
+}
+
+const getTimeAgoColor = (timestamp: string | null): string => {
+  if (!timestamp) return 'text.secondary';
+  const daysDifference = dayjs().diff(dayjs(timestamp), 'day');
+  if (daysDifference < 3) return 'green';
+  if (daysDifference <= 5) return 'orange';
+  return 'red';
+};
+
+const getResourceIcon = (name: ResourceName) => {
+  switch (name) {
+    case ResourceName.COURSE_NOTES:
+      return <ArticleIcon />;
+    case ResourceName.COURSE_QUIZ:
+      return <QuizIcon />;
+    case ResourceName.COURSE_COMMENTS:
+      return <CommentIcon />;
+    case ResourceName.COURSE_STUDY_BUDDY:
+      return <Diversity3Icon />;
+    case ResourceName.COURSE_HOMEWORK:
+      return <GradingIcon />;
+    case ResourceName.COURSE_ACCESS:
+      return <LockOpenIcon />;
+    default:
+      return name[0];
+  }
+};
+
+async function getLastUpdatedQuiz(
+  courseId: string
+): Promise<{ description: string | null; timeAgo: string | null; timeStamp: string | null }> {
+  try {
+    const quizList = await getCourseQuizList(courseId);
+    const timeStamp = quizList[quizList.length - 1].quizStartTs;
+    const description = `Last Quiz: ${dayjs(timeStamp).format('YYYY-MM-DD')}`;
+    const timeAgo = calculateTimeAgo(timeStamp);
+    return { description, timeAgo, timeStamp };
+  } catch (error) {
+    console.error('Error fetching course data:', error);
+    return { description: null, timeAgo: null, timeStamp: null };
+  }
+}
+
+async function getLastUpdatedHomework(
+  courseId: string
+): Promise<{ description: string | null; timeAgo: string | null; timeStamp: string | null }> {
```

Comment:
Time stamp is a single word. No need to upper case `S`. 
```suggestion
): Promise<{ description: string | null; timeAgo: string | null; timestamp: string | null }> {
```